README.md

```diff
@@ -2,3 +2,17 @@
 
 [coredns](https://github.com/coredns/coredns) Builds specific to [Kuma](https://github.com/kumahq/kuma).
 
+## Making dependabot work 
+
+To be able to rely on dependabot we have a `go mod` in the root with the 2 dependencies we have. Because `go mod tidy` will remove
+unused dependencies we have an `internal` package that imports these 2 dependencies.
+
+## Building
+
+You can build the entire package with `make tar` everything will then be in `build/out`. 
+In practice this is all done in a github action.
+
+## Releasing a new version
+
+Once everything is updated just push a tag and github actions will create a release with the binaries.
```

Comment:
I think this workflow would be a good opportunity to avoid the "tag first, then build (maybe, cause it might not succeed)" paradigm and instead just launch the workflow with a given version and have it create a tag/release at the end only if it's successful.

Reply:
What's wrong with just repushing the tag if it didn't work?

Reply:
I think we had this discussion at some point but I think generally the more immutable tags are, the better.

Or otherwise, _why_ even use a tag as a trigger in the first place? Why should the tag exist before the release does and not just have them both pushed at the same time? Why not just directly _trigger the workflow_ without the tag in between?